app/Repositories/PostRepository.php

```diff
@@ -170,6 +170,54 @@ public function reviews($data)
     }
 
     /**
+     * Updates a post's tags when added or deleted.
+     *
+     * @param array $data
+     *
+     * @return
+     */
+    public function tag($data)
+    {
+        $post = Post::where(['id' => $data['post_id']])->first();
+
+        // Check if the post already has the tag.
+        // If so, soft delete. Otherwise, add the tag to the post.
+        if (in_array($data['tag_name'], $post->tagNames())) {
+            $post->untag($data['tag_name']);
+
+            // Create an event when the tag is deleted.
+            if (!in_array($data['tag_name'], $post->tagNames())) {
+                $post->tag_name = $data['tag_name'];
+                $post->admin_northstar_id = $data['admin_northstar_id'];
+                $post->action = 'delete';
+
+                $event = Event::create([
```

Comment:
@chloealee So I was thinking something like this. I just tried this out on my local and it seems to work but let me know if something doesn't look right here. 

```

        Tagged::saved(function($tagged) {
            $post = Post::find($tagged->taggable_id);
            $adminId = auth()->user()->northstar_id;

            $post->action = [
                'type' => 'tagged',
                'admin' => $adminId,
            ];

            Event::create([
                'eventable_id' => $post->id,
                'eventable_type' => 'Conner\Tagging\Model\Tagged',
                'content' => $post->toJson(),
            ]);
        });
```